### PR TITLE
lookup_linke_turbidity speed improvement, mat to h5

### PR DESCRIPTION
**Pull request by cedricleroy**
- [x] Closes issue #437
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

**Note:**  `git diff upstream/master -u -- "*.py" | flake8 --diff` is giving me `fatal: bad revision 'upstream/master'`. I use `pylint` though and it didn't complain on additions.

### CODE REVIEW

File: docs/sphinx/source/whatsnew/v0.5.2.rst [hunk 1]
```diff
@@ -9,7 +9,7 @@ API Changes
 
 Enhancements
 ~~~~~~~~~~~~
-*
+* Improve clearsky.lookup_linke_turbidity speed (IO) (:issue:`437`)
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.5.2.rst]: Let's add the ``.mat`` to ``.h5`` change to the API changes section.

File: pvlib/clearsky.py [hunk 2]
```diff
@@ -214,7 +211,14 @@ def lookup_linke_turbidity(time, latitude, longitude, filepath=None,
         np.around(_linearly_scale(longitude, -180, 180, 0, 4320))
         .astype(np.int64))
 
-    lts = linke_turbidity_table[latitude_index][longitude_index]
+    lt_h5_file = tables.open_file(filepath)
+    try:
+        lts = lt_h5_file.root.LinkeTurbidity[latitude_index, longitude_index, :]
+    except IndexError:
+        raise IndexError('Latitude should be between 90 and -90, '
```
Comment (wholmgren) [inline pvlib/clearsky.py]: We should add a test that asserts that this exception is raised under invalid inputs.

Also, I confirmed that the code below does in fact execute the finally block despite a new exception being raised.

```python
try:
    raise IndexError
except IndexError:
    raise IndexError('blah')
finally:
    print('finally')
```

Comment (cedricleroy) [reply pvlib/clearsky.py]: I actually did write a test for that, but then realized it was [already covered](https://github.com/pvlib/pvlib-python/blob/master/pvlib/test/test_clearsky.py#L479).

For the `try except finally`, I was hoping `open_file` to come with a context manager, but it does not look like it is the case and fall back to the old school way to make sure the file get closed whatever happen.

**Comment by cedricleroy** [comment]
Looks like the build is passing now. I can squash commits if needed.

File: pvlib/clearsky.py (hunk 2 above)
Comment (wholmgren) [reply pvlib/clearsky.py]: my bad, thanks for reminding me...

**Comment by wholmgren** [comment]
Looks finished to me. I plan to merge tomorrow, so now is the time to comment if anyone has objections.

**Pull request merged**